rest-service/manager_rest/dsl_functions.py

```diff
@@ -246,6 +311,41 @@ def check_vault_connection(url, token, path):
     return vault_status
 
 
+def check_cloudify_connection(**connection_parameters):
+    status = {
+        'status': True,
+        'message': '',
+    }
+
+    if not connection_parameters.get('protocol'):
+        connection_parameters['protocol'] = 'https'
+
+    if cert := connection_parameters.get('cert'):
+        temp_cert_file = _get_cloudify_cert_temp_file(cert)
+        connection_parameters['cert'] = temp_cert_file.name
+
+    client = CloudifyClient(
+        **connection_parameters,
+    )
+
+    try:
+        status_response = client.manager.get_status()
+        status_response.raise_for_status()
```

Comment:
Nope! `.get_status()` returns a dict. No need for `raise_for_status`, because the restclient will already throw if it receives a non-200 here.

For reference (pardon my trust-all in the example),
```python
>>> c = CloudifyClient(username='admin', password='admin', host='127.0.0.1', protocol='https', trust_all=True)
>>> c.manager.get_status()
/opt/manager/env/lib/python3.10/site-packages/urllib3/connectionpool.py:1045: InsecureRequestWarning: Unverified HTTPS request is being made to host '127.0.0.1'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
{'status': 'OK', 'services': {'Webserver': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'AMQP-Postgres': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Management Worker': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Manager Rest-Service': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Cloudify API': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Cloudify Execution Scheduler': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Cloudify Console': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'PostgreSQL': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'RabbitMQ': {'status': 'Active', 'is_remote': False, 'extra_info': {'connection_check': 'OK'}}, 'Cloudify Composer': {'status': 'Active', 'is_remote': False, 'extra_info': {}}, 'Monitoring Service': {'status': 'Active', 'is_remote': False, 'extra_info': {}}}}
>>> 
```